File: 2021/02/cpp/pt2.cpp
#include <fstream>
#include <iostream>
#include <string>
using namespace std;

int main()
{
    fstream file("../input.txt");
    string commandRaw;

    int hor = 0;
    int depth = 0;
    int aim = 0;

    while (getline(file, commandRaw))
    {
        string command = commandRaw.substr(0, commandRaw.find(' '));
        int num = stoi(commandRaw.substr(commandRaw.find(' ')));

        if (command == "up")
        {
            aim -= num;
        }
        else if (command == "down")
        {
            aim += num;
        }
        else if (command == "forward")
        {
            hor += num;
            depth += aim * num;
        }
    }

    cout << hor * depth << '\n';
}

File: 2021/02/python/pt2.py
import os
with open(f'{os.path.dirname(__file__)}/../input.txt', 'r') as file:
    commands = file.read().splitlines()

hor = depth = aim = 0
for i in commands:
    command, num = i.split()
    num = int(num)
    if command == "up":
        aim -= num
    elif command == "down":
        aim += num
    elif command == "forward":
        hor += num
        depth += aim * num

print(hor * depth)
